source/dem/dem.cc

```diff
@@ -254,6 +258,11 @@ DEMSolver<dim>::load_balance()
 
   boundary_cell_object.build(triangulation, parameters.floating_walls);
 
+  if (parameters.grid_motion.motion_type !=
+      Parameters::Lagrangian::GridMotion<dim>::MotionType::none)
+    boundary_cell_object.update_boundary_info_after_grid_motion(
+      updated_boundary_points_and_normal_vectors);
+
```

Comment:
Why do you need to make the update during the load balance?

Reply:
When we call load-balancing, all the structures related to triangulation including finding cell neighbors and boundary cells should be rebuilt.